src/components/CurrentScenarioSelector/CurrentScenarioSelector.js

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { HierarchicalComboBox } from '@cosmotech/ui';
+import { useSortedScenarioList } from '../../hooks/ScenarioListHooks';
+import { STATUSES } from '../../state/commons/Constants';
+import { useCurrentScenarioData, useCurrentScenarioStatus, useFindScenarioById } from '../../state/hooks/ScenarioHooks';
+import { t } from 'i18next';
+
+const CurrentScenarioSelector = ({ disabled, renderInputToolType }) => {
+  const sortedScenarioList = useSortedScenarioList();
+  const currentScenarioData = useCurrentScenarioData();
+  const currentScenarioStatus = useCurrentScenarioStatus();
+
+  const changeScenario = useFindScenarioById();
+
+  const handleScenarioChange = useCallback(
+    (event, scenario) => {
+      changeScenario(scenario.id);
+    },
+    [changeScenario]
+  );
+
+  const noScenario = currentScenarioData === null;
+  const scenarioListDisabled = sortedScenarioList.length === 0 || noScenario;
+  const isSwitchingScenario = currentScenarioStatus === STATUSES.LOADING;
+
+  const scenarioListLabel = noScenario ? null : t('views.scenario.dropdown.scenario.label', 'Scenario');
+  const scenarioValidationStatusLabels = {
+    rejected: t('views.scenario.validation.rejected', 'Rejected'),
+    validated: t('views.scenario.validation.validated', 'Validated'),
+  };
+
+  const hierarchicalComboBoxLabels = {
+    label: scenarioListLabel,
+    validationStatus: scenarioValidationStatusLabels,
+  };
+
+  return (
+    <HierarchicalComboBox
+      value={currentScenarioData}
+      values={sortedScenarioList}
+      labels={hierarchicalComboBoxLabels}
+      renderInputToolType={renderInputToolType}
+      disabled={disabled || scenarioListDisabled || isSwitchingScenario}
+      handleChange={handleScenarioChange}
+    />
+  );
+};
+
+CurrentScenarioSelector.propTypes = {
+  /**
+   * Component's labels:
+   * Structure:
+   * <pre>
+     {
+       label: 'string',
+       validationStatus:
+       {
+         rejected: 'string',
+         validated: 'string',
+       }
+     }
+   * </pre>
+   */
+  labels: PropTypes.shape({
+    label: PropTypes.string,
+    validationStatus: PropTypes.shape({
+      rejected: PropTypes.string,
+      validated: PropTypes.string,
+    }),
+  }),
+  /**
+   *  Define the HierarchicalComboBox's state:
+   *  - true : the button is disabled (the tooltip will guide users on how to enable the button)
+   *  - false : the button is enabled
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Tooltip on HierarchicalComboBox
+   */
+  renderInputToolType: PropTypes.string,
```

Comment:
Typo: `renderInputToolType` should be `renderInputTooltip`

Reply:
done